sdk/analytics/src/main/java/com/klaviyo/analytics/networking/ApiClient.kt

```diff
@@ -27,4 +33,18 @@ internal interface ApiClient {
      * @param profile
      */
     fun enqueueEvent(event: Event, profile: Profile)
+
+    /**
+     * Register an observer to be notified when an API request is enqueued or changes state
+     *
+     * @param observer
+     */
+    fun onApiRequest(observer: ApiObserver)
```

Comment:
I opted to just expose the more complicated "events" as part of the services they belong to, rather than in the Log interface. Some of my services already had this pattern anyway.